src/compile/data/parse.ts

```diff
@@ -384,13 +399,41 @@ export function parseData(model: Model): DataComponent {
     head = StackNode.makeFromEncoding(head, model) ?? head;
   }
 
+  let preFilterInvalid: OutputNode | undefined;
```

Comment:
**TODO: file a separate ticket** 

Separate from this PR, I have a hunch that this will remain slightly incorrect for stacked chart, since null values will be "stacked", but it's probablly